test/utils/test_deserialization.py

```diff
@@ -132,3 +134,39 @@ def test_chat_generator_no_from_dict_method(self):
         data = {"chat_generator": chat_generator.to_dict()}
```

Comment:
Once we've updated `deserialize_chatgenerator_inplace` to call `deserialize_component_inplace` then we should remove the unnecessary tests.